packages/core/src/llm/utils.ts

```diff
@@ -8,3 +6,19 @@ export async function* streamConverter<S, D>(
     yield converter(data);
   }
 }
+
+export async function* streamReducer<S, D>(params: {
```

Comment:
@himself65 @yisding I wrote these streaming helpers - they simplify much of our streaming code. Do you know a library that we can use to deal with this concern?

Reply:
nodejs utils might have such thing but we shouldn't use that

Reply:
ok, then we just leave it as is for now. Please have a look, those helpers are also useful for adding new LLMs